From 492fdce2626efc01004bea4aed4ea89b7c5cb5b3 Mon Sep 17 00:00:00 2001
From: Kiran Gunda <kgunda@codeaurora.org>
Date: Thu, 17 May 2018 17:00:34 +0530
Subject: regulator: core: Fix buffer overflow issue

There is a possible buffer overflow in the "reg_debug_volt_get"
function. Fix it by passing the appropriate byte count.

Change-Id: I30868790c42cdb225af74054532a75c49506fe82
Signed-off-by: Kiran Gunda <kgunda@codeaurora.org>
---
 drivers/regulator/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index 7b2a4b6d..3bfc88b 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -3824,7 +3824,7 @@ static ssize_t reg_debug_volt_get(struct file *file, char __user *buf,
 	mutex_lock(&debug_buf_mutex);
 
 	output = snprintf(debug_buf, MAX_DEBUG_BUF_LEN-1, "%d\n", voltage);
-	rc = simple_read_from_buffer((void __user *) buf, output, ppos,
+	rc = simple_read_from_buffer((void __user *) buf, count, ppos,
 					(void *) debug_buf, output);
 
 	mutex_unlock(&debug_buf_mutex);
-- 
cgit v1.1

